doc/contributors/design-proposals/pull-bundle-on-a-cluster.md

```diff
@@ -0,0 +1,127 @@
+# Pull Bundle on a Cluster
+
+The purpose of this proposal is to allow bundle data to be pulled at install time. At a high level, this starts with launching the bundle inside a pod and then writing the data found inside in the container into a configmap.
+
+## Launching a bundle image for data extraction
+
+An API will be made that handles launching a job. The only required parameter will be specifying the bundle image.
+
+The job spec will look similar to the following:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: deploy-bundle-image
+spec:
+  #ttlSecondsAfterFinished: 0 # alpha feature - https://github.com/kubernetes/enhancements/issues/592, https://github.com/kubernetes/kubernetes/pull/82082
+  template:
+    metadata:
+      name: bundle-image
+    spec:
+      containers:
+      - name: bundle-image
+        image: &image bundle-image
+        command: ['/injected/operator-registry', 'serve']
+        env:
+          - name: CONTAINER_IMAGE
+            value: *image
+        volumeMounts:
+        - name: copydir
+          mountPath: /injected
+      initContainers:
+      - name: copy-binary
+        image: init-operator-manifest
+        imagePullPolicy: Never
+        command: ['/bin/cp', '/operator-registry', '/copy-dest']
+        volumeMounts:
+        - name: copydir
+          mountPath: /copy-dest
+      volumes:
+      - name: copydir
+        emptyDir: {}
+      restartPolicy: OnFailure
+```
+
+## Serving the bundle data
+
+New code will be made in operator-registry to provide functionality for traversing the directies of the bundle image for writing to a configmap (example above is "operator-registry serve"), the format for which is discussed in more detail below. The configmap will also use a generated name to avoid collisions and will be labeled to match the requested bundle image. It will be the responsibility of the caller to 1) launch the job, 2) watch for the target configmap to be created 3) to delete the job (until ttlSecondsAfterFinished is available) 4) delete the configmap after reading.
```

Comment:
Alternatively, instead of generating a name for the configmap, could use a sanitized version of the bundle image name and use that (can't use colons, but maybe that's the only unacceptable character. also 253 character limit might be worth worrying about). Either way, I don't think specifying the name of the configmap by the caller is the correct approach.

Reply:
Hrm, labels are actually restricted to 63 characters. Docker registry names are limited to 256 characters. And quay names seem to be even potentially longer (https://coreos.com/quay-enterprise/releases/#2.0.4).

Reply:
You could always create a shortened hash of the fully qualified image name -- that would help you meet both the length and character-set restrictions.

Reply:
That's a good idea, however, I pivoted a little bit.
I ended up retracting my original aversion to specifying a configmap, at least within the bundle image. Rather than worrying about character limits I ended up putting the configmap creation code in the "launch" function, so that the caller doesn't have to worry about configmap naming. I'll be updated the proposal here shortly that hopefully fully clarifies the new/final direction.